### PR TITLE
Reverted Google auth functionality for legacy use

### PR DESCRIPTION
**Pull request by EvanBacon**
Because we are not focusing on adding Firebase authentication, we should revert this module to use the old Google API console.
Google auth being a part of Firebase was important for using firebase invites, which is now deprecated 🙃
This revert will account for users working in Expo vs unmanaged/standalone. There will be some repetitive functionality but considering how many issues have been opened, I think it'll be fine.

# Test Plan

**`androidStandaloneAppClientId` & `iosStandaloneAppClientId`**

Using in ejected Expo should have the same functionality as a standalone app. ie: Bundle ID & Package will be custom and should be registered by the app.json before ejecting.

**`androidClientId` & `iosClientId`**

Create a new project in Google API (not Firebase) and use `host.exp.Exponent` as the bundle ID and package.

### CODE REVIEW

**Comment by EvanBacon** [comment]
Once I get approval on the docs, I'll retroactively apply them.

**Comment by EvanBacon** [comment]
Assumes this is merged: #3402

File: packages/expo/src/Google/Google.ts
Comment (ide) [inline packages/expo/src/Google/Google.ts]: lol name this a bit differently:
```suggestion
  if (!validation.isValid) {
```

File: packages/expo/src/Google/Google.ts
Comment (ide) [inline packages/expo/src/Google/Google.ts]: `Array(PROJECT_ID_LENGTH).fill('x')` seems simpler unless you want the XxXxX for a specific reason

File: packages/expo/src/Google/Google.ts
Comment (EvanBacon) [reply packages/expo/src/Google/Google.ts]: woopsies

File: packages/expo/src/Google/Google.ts
Comment (ide) [inline packages/expo/src/Google/Google.ts]:
```suggestion
      'Keys for Android and iOS cannot be the same value. Ensure you are linking the client IDs matching the given platforms in the Google APIs console: https://console.developers.google.com/apis/credentials'
```

File: packages/expo/src/Google/Google.ts
Comment (ide) [inline packages/expo/src/Google/Google.ts]:
```suggestion
export type GoogleLogInConfig = {
```

**Review by ide — APPROVED** [review]

**Comment by mkyo2012** [comment]
problem solved?

**Pull request merged**
